### PR TITLE
Truncate JNI boolean returns

### PR DESCRIPTION
**Pull request by tajila**
Truncate JNI boolean returns

Signed-off-by: tajila <atobia@ca.ibm.com>

### CODE REVIEW

File: runtime/oti/VMHelpers.hpp [hunk 1]
```diff
@@ -1381,7 +1381,7 @@ class VM_VMHelpers
 			}
 			break;
 		case J9NtcBoolean:
-			*returnStorage = (UDATA)(U_8)*returnStorage;
+			*returnStorage = (UDATA)(U_8)(*returnStorage & 1);
```
Comment (DanHeidinga) [inline runtime/oti/VMHelpers.hpp]: It's clearer if the `& 1` is bubbled out of the read/cast brackets: `((UDATA)(U_8)*returnStorage) & 1;`

**Comment by tajila** [comment]
@DanHeidinga I've made the change

**Comment by tajila** [comment]
@andrewcraik The JIT will need similar changes. Likewise with unsafe boolean stores

**Comment by DanHeidinga** [comment]
Discussed with @andrewcraik & @gacholio -  the prefered approach is to have the JIT update the code generated for the directToJNI case to include the `& 1` as this covers all cases and adds minimal instructions only in the boolean return case.

**Pull request merged**